client/cluster_test.go

```diff
@@ -124,7 +124,7 @@ func TestAddMachine(t *testing.T) {
 
 	latestImageMaster := api.LatestImage(cloudProvider, api.Master)
 	latestImageWorker := api.LatestImage(cloudProvider, api.Master)
-	firstImageMaster := cloudProvider.MachineImages(api.Master)[0]
+	firstImageMaster := cloudProvider.MachineImages(api.Master)[2]
```

Comment:
Does this needs to be latest image - 1? If so would it make sense to do:
```suggestion
  allImages = cloudProvider.MachineImages(api.Master)
  firstImageMaster := allImages[ len(allImages) - 2 ]
```

Reply:
Yes that looks much better! I guess it doesn't *have* to be latest - 1 if there are multiple with the same minor kubernetes version, but the point of the test (I think) is just to see that something other than latest also works.